Lps 88817 lps 94946 search remove

oliv-yu opened the pull request:
The first commit added the new blue toolbar, fixed a bug with added result's dropdown ellipses showing, and it now positions pinned items to the top of the list.

The second commit removes all Searchbar related stuff :'(

The third commit, I renamed FilterDisplay lol.

thektan commented on modules/apps/portal-search/portal-search-ranking-web/src/main/resources/META-INF/resources/js/components/list/Item.es.js:
```diff
@@ -563,13 +563,15 @@ class Item extends PureComponent {
+							onClickPin={this._handlePin}
+							pinned={pinned}
+						/>
+					}
```
Can we do this check inside `ItemDropdown` itself? If `onClickHide` isn't available, we shouldn't hide this whole dropdown since `onClickPin` should still be possible.

oliv-yu commented:
New stuff!
- Added changes for above comments
- FilterDisplay, FilterInput tests
- Removed old properties from component tests
- Removed hiding from addResults 
- Add explanations in certain places

thektan commented on modules/apps/portal-search/portal-search-ranking-web/src/main/resources/META-INF/resources/js/components/ResultsRankingForm.es.js:
```diff
@@ -444,30 +432,47 @@ class ResultsRankingForm extends Component {
-				const mappedData = items ? resultsDataToMap(newItems) : {};
+				const definedItems = items ? items : {};
+
+				// Record duplicate results in order to set addedResult property to
```
Can we use "get" instead of "record"? "Record" seems like the value is being saved to someplace. 

```suggestion
				// Get duplicate results in order to set addedResult property to
```

thektan commented on modules/apps/portal-search/portal-search-ranking-web/src/main/resources/META-INF/resources/js/components/ResultsRankingForm.es.js:
```diff
@@ -357,23 +317,37 @@ class ResultsRankingForm extends Component {
-				const mappedData = items ? resultsDataToMap(newItems) : {};
+				const definedItems = items ? items : {};
+
+				// Record duplicate results in order to set addedResult property to
```
```suggestion
				// Get duplicate results in order to set addedResult property to
```

thektan commented on modules/apps/portal-search/portal-search-ranking-web/src/main/resources/META-INF/resources/js/components/ResultsRankingForm.es.js:
```diff
@@ -444,30 +432,47 @@ class ResultsRankingForm extends Component {
-					[];
 
-				const mappedData = items ? resultsDataToMap(newItems) : {};
+				const definedItems = items ? items : {};
```
We can use a short circuit instead.

```suggestion
				const definedItems = items || {};
```

thektan commented on modules/apps/portal-search/portal-search-ranking-web/src/main/resources/META-INF/resources/js/components/ResultsRankingForm.es.js:
```diff
@@ -477,10 +482,22 @@ class ResultsRankingForm extends Component {
-								...state.dataMap,
+
+								// Set the addedResult to false if it shows up as one
+								// of the results that pops up in search.
```
The wording of this comment is a bit strange since we're saying "if" something shows up when in reality we're setting all duplicate items to false? 

Can we update the comments to be more descriptive? I think it's important to explain **why** we're setting these duplicate items to false since it's not clear how fetching items would need to set `addedResult` to false.

thektan commented on modules/apps/portal-search/portal-search-ranking-web/src/main/resources/META-INF/resources/js/components/ResultsRankingForm.es.js:
```diff
@@ -614,6 +620,9 @@ class ResultsRankingForm extends Component {
 		const mappedData = resultsDataToMap(addedResultsDataList);
 
+		// Make sure that all added results that don't overlap with the original
+		// search results get marked as pinned and addedResult.
```
Instead of "overlap" maybe something along the lines of "included with the original search results" or "part of"? "Overlap" seems to be like we're describing something visual.
